BlueDBOnDisk/src/main/java/org/bluedb/disk/lock/Lock.java

```diff
@@ -35,4 +52,9 @@ public void lockAndUnlockForWrite() {
 		// We need a lock that we know is in the map and that we got the lock before another thread.
 		unlockForWrite();
 	}
+

```

Comment:
Ya, probably not the best thing, but it allows use to get rid of the comments and instantly have a bluedb jar that tells us exactly what is happening with the locks. Its super verbose, but it is the way I was able to trace what exactly was happening so that I could fix this bug. Figured it would be nice to be able to easily turn that output back on without taking a resource hit in production or having to cover the code with tests. What do you think? Any ideas how to do this better right now? I don't think a full logging overhaul is worth it right now.

Reply:
Oh, actually, my reply is about the file in general. But these particular comments are your comments from before my refactor when I created the Lock class. You were explaining why we were locking and unlocking the write lock here. It is a little more clear after my refactor but I figured that I'd keep the comment in here since there is some good information in here that can help someone understand how the locking works. It is a little hard to understand why the `lockAndUnlockForWrite` method is needed.